### PR TITLE
edits to the import function

### PR DESCRIPTION
**Pull request by Farnazmdi**
I added a test function to the `test_import_lineage` and now I am only importing the first file (of the three files) of EGF data that fails.
Here is the error:

`>       output = run_Analyze_over(dataFull, desired_num_states, atonce=True)

lineage/tests/test_import_lineage.py:75: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
lineage/Analyze.py:78: in run_Analyze_over
    output = [prom.result() for prom in prom_holder]
lineage/Analyze.py:78: in <listcomp>
    output = [prom.result() for prom in prom_holder]
/usr/lib/python3.9/concurrent/futures/_base.py:445: in result
    return self.__get_result()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = None

    def __get_result(self):
        if self._exception:
            try:
>               raise self._exception
E               FloatingPointError: invalid value encountered in true_divide

/usr/lib/python3.9/concurrent/futures/_base.py:390: FloatingPointError
`

### CODE REVIEW

**Comment by Farnazmdi** [comment]
The more accurate place or error is this:

Traceback (most recent call last):
  File "/usr/lib/python3.9/concurrent/futures/process.py", line 243, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/home/farnazm/tHMM/lineage/Analyze.py", line 28, in Analyze_list
    _, _, _, _, LL = fit_list(tHMMobj_list)
  File "/home/farnazm/tHMM/lineage/tHMM.py", line 223, in fit_list
    MSD_list, NF_list, betas_list, gammas_list = map(list, zip(*[do_E_step(tHMM) for tHMM in tHMMobj_list]))
  File "/home/farnazm/tHMM/lineage/tHMM.py", line 223, in <listcomp>
    MSD_list, NF_list, betas_list, gammas_list = map(list, zip(*[do_E_step(tHMM) for tHMM in tHMMobj_list]))
  File "/home/farnazm/tHMM/lineage/BaumWelch.py", line 36, in do_E_step
    get_nonleaf_NF_and_betas(tHMMobj, MSD, EL, NF, betas)
  `File "/home/farnazm/tHMM/lineage/UpwardRecursion.py", line 203, in get_nonleaf_NF_and_betas
    betas[num][pii, :] = fac1 / NF[num][pii]`
FloatingPointError: invalid value encountered in true_divide

**Comment by Farnazmdi** [comment]
I caught a mistake in the importing function in the order of cells... I think there is no point in looking into this before I fix that..

**Comment by Farnazmdi** [comment]
right now, we are ignoring the lineages that have 1 or 3 children, and for those cells that appear at the very end, I set their lifetime = 0.5 instead of 0.

File: lineage/tests/test_import_lineage.py [hunk 1]
```diff
@@ -62,3 +64,12 @@ def test_MCF10A(self):
         np.testing.assert_allclose(lin1[0].obs, [1, 30.0, 1], rtol=1e-2)
         np.testing.assert_allclose(lin1[1].obs, [np.nan, 17.5, 1], rtol=1e-2)
         np.testing.assert_allclose(lin1[2].obs, [np.nan, 17.5, 1], rtol=1e-2)
+
+    def test_bic(self):
```
Comment (aarmey) [inline lineage/tests/test_import_lineage.py:68]: Do you know why the test is failing?

File: lineage/import_lineage.py [hunk 2]
```diff
@@ -185,7 +172,10 @@ def assign_observs_MCF10A(cell, lineage: list, uniq_id: int):
 
     # cell's lifetime
     cell.obs[1] = (np.max(lineage.loc[lineage['TID'] == uniq_id]['tmin']) - np.min(lineage.loc[lineage['TID'] == uniq_id]['tmin'])) / 60
-
+    cell.obs[3] = np.mean(lineage.loc[lineage['TID'] == uniq_id]['average_velocity'])
+    cell.obs[4] = np.mean(lineage.loc[lineage['TID'] == uniq_id]['distance_mean'])
+    if cell.obs[1] <= 0.1:
+        cell.obs[1] = 0.5
```
Comment (aarmey) [inline lineage/import_lineage.py:178]: Is this necessary? Why are there cells with lifetimes less than 0.1?

File: lineage/import_lineage.py [hunk 3]
```diff
@@ -199,17 +189,17 @@ def MCF10A(condition: str):
 
     elif condition == "EGF":
         data1 = import_MCF10A("lineage/data/MCF10A/EGF_1.csv")
-        # data2 = import_MCF10A("lineage/data/MCF10A/EGF_2.csv")
```
Comment (aarmey) [inline lineage/import_lineage.py:202]: Is it possible to import these files but skip the lineages with bad data?

**Review by aarmey — CHANGES_REQUESTED** [review]

File: lineage/import_lineage.py (hunk 2 above)
Comment (Farnazmdi) [reply lineage/import_lineage.py]: There were the cells with 0 lifetime, I fixed it now.

File: lineage/import_lineage.py (hunk 3 above)
Comment (Farnazmdi) [reply lineage/import_lineage.py:202]: Yes, it is done in this pr.

File: lineage/tests/test_import_lineage.py (hunk 1 above)
Comment (Farnazmdi) [reply lineage/tests/test_import_lineage.py:68]: oh.. I have added the distance and velocity to the observations and I forgot to apply those changes in the test. I fixed it.

**Review by aarmey — APPROVED** [review]
Great. Merging.

**Pull request merged**